processors/ratelimiter/ratelimiterprocessor.go

```diff
@@ -0,0 +1,95 @@
+package ratelimiter
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	pb_struct "github.com/envoyproxy/go-control-plane/envoy/extensions/common/ratelimit/v3"
+	pb "github.com/envoyproxy/go-control-plane/envoy/service/ratelimit/v3"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+	"google.golang.org/grpc/metadata"
+)
+
+type processor struct {
+	rateLimitServiceClient   pb.RateLimitServiceClient
+	domain                   string
+	domainSoftLimitThreshold uint32
+	logger                   *zap.Logger
+	tenantIDHeaderName       string
+}
+
+const (
+	TenantID       = "tenant_id"
+	GlobalTenantID = "global_tenant_id"
+)
+
+// ProcessTraces implements processorhelper.ProcessTracesFunc
+func (p *processor) ProcessTraces(ctx context.Context, traces ptrace.Traces) (ptrace.Traces, error) {
```

Comment:
You might want to implement ConsumeTraces actually. That way you can return nil if you want to drop.

Reply:
Updated